Bottom: 143c699a373dc8318cadc73f45fe932ee2c97b48
Top:    c8ca7d41c148c6162cd0bb7d77cff8872a526e42
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-05-08 12:05:24 +0100

Refresh of add-action-to-set-netclass

---

diff --git a/src/netlist.c b/src/netlist.c
index 7114ae0..17b9957 100644
--- a/src/netlist.c
+++ b/src/netlist.c
@@ -241,14 +241,12 @@ netlist_style (LibraryMenuType *net, const char *style)
   net->Style = STRDUP ((char *)style);
 }
 
-#if 0
 static void
 netlist_netclass (LibraryMenuType *net, const char *netclass)
 {
   free (net->Netclass);
   net->Style = STRDUP ((char *)netclass);
 }
-#endif
 
 /* The primary purpose of this action is to rebuild a netlist from a
    script, in conjunction with the clear action above.  */
@@ -398,6 +396,8 @@ Netlist (int argc, char **argv, Coord x, Coord y)
     }
   else if (strcasecmp (argv[0], "style") == 0)
     func = (NFunc)netlist_style;
+  else if (strcasecmp (argv[0], "class") == 0)
+    func = (NFunc)netlist_netclass;
   else if (strcasecmp (argv[0], "add") == 0)
     {
       /* Add is different, because the net/pin won't already exist.  */
@@ -504,6 +504,10 @@ Netlist (int argc, char **argv, Coord x, Coord y)
 	{
 	  netlist_style (net, ARG(2));
 	}
+      else if (func == (void *)netlist_netclass)
+	{
+	  netlist_netclass (net, ARG(2));
+	}
       else if (argc > 2)
 	{
 	  int l = strlen (argv[2]);
@@ -513,7 +517,7 @@ Netlist (int argc, char **argv, Coord x, Coord y)
 		    && net->Entry[j].ListEntry[l] == '-'))
 	      {
 		pin = net->Entry + j;
-		pin_found = 1;
+	pin_found = 1;
 		func (net, pin);
 	      }
 	}
